components/abstractions/src/main/java/com/microsoft/kiota/store/InMemoryBackingStore.java

```diff
@@ -134,7 +141,20 @@ private Object getValueFromWrapper(final String entryKey, final Pair<Boolean, Ob
     @Nullable public <T> T get(@Nonnull final String key) {
         Objects.requireNonNull(key);
         final Pair<Boolean, Object> wrapper = this.store.get(key);
-        final Object value = this.getValueFromWrapper(key, wrapper);
+        if (wrapper == null) {
+            return null;
+        }
+        final Object value = this.getValueFromWrapper(wrapper);
+
+        boolean hasChanged = wrapper.getValue0();
+        if (getReturnOnlyChangedValues() && !hasChanged) {
+            ensureCollectionPropertiesAreConsistent();
```

Comment:
I'm worried about the performance of enumerating everything on every get. Can you please provide more context on why this is required here?

Reply:
If a nested collection is updated directly, the backing store has no way of being notified of this change e.g.
```java
backedModel.getAdditionalData().put("newKey", "newVal");
```
We need to run through the hierarchy of `BackedModel`s looking for any collection values & check their sizes.

Yes, this is expensive but:
- These checks are limited to collection properties that are NOT dirty since they may have gotten an addition/removal directly. 
- This check should happen at most once on each `BackedModel` in a hierarchy. Once a collection property is marked as dirty by these checks, future get()'s on the same `BackedModel` should skip this.

I can further limit this by doing this only if the value for the `key` is a collection.

Reply:
thanks for providing more information here. Yes, let's try to reduce iterations as much as possible.